### ISSUE TITLE
Create Async port

### ISSUE THREAD

**Issue by johnelse** [johnelse/ocaml-irc-client#2]
Async should be an optional dependency.

**Comment by cfcs** [comment]
If you're going to support Async / Lwt, please see `conduit`:
https://github.com/mirage/ocaml-conduit/

**Comment by johnelse** [comment]
Indeed, I'd like to switch to conduit when I have some spare time. I'm also considering dropping the Unix variant, as everyone I've spoken to seems to be using Lwt :)
